ood-portal-generator/lib/ood_portal_generator.rb

```diff
@@ -26,5 +29,25 @@ def scl_apache?
       return false if ("#{env['ID']} #{env['ID_LIKE']}" =~ /rhel/ && env['VERSION_ID'] =~ /^8/)
       true
     end
+
+    def fqdn
+      Socket.gethostbyname(Socket.gethostname).first
+    end
+
+    def dex_user
+      Etc.getpwnam('ondemand-dex')
+      'ondemand-dex'
+    rescue ArgumentError
+      Etc.getlogin
+    end
+
+    def dex_group
+      Etc.getgrnam('ondemand-dex')
+      'ondemand-dex'
```

Comment:
Same as above, we have `Etc.getgrnam('ondemand-dex')` and `'ondemand-dex'`.

Reply:
The execution of `Etc.getgrnam` is just to see if `ondemand-dex` group exists so that a sane default can be chosen if not.

Reply:
Ah. Maybe then `Etc.getgrnam('ondemand-dex').name` and a method description e.g. return ondemand-dex if exists, else user group name

Reply:
Updated with shorter alternative and also documented both user/group functions.